apps/constellation/constellation.cpp

```diff
@@ -92,6 +95,21 @@ uint16_t LookupLocalPort(Manifest const &manifest, ServiceType service, uint16_t
   return manifest.GetLocalPort(identifier);
 }
 
+std::shared_ptr<ledger::DAGInterface> GenerateDAG(bool generate, std::string const &db_name,
+                                                  bool                          load_on_start,
+                                                  Constellation::CertificatePtr certificate)
+{
+  std::shared_ptr<ledger::DAGInterface> ret;
+
+  if (generate)
+  {
+    ledger::DAG *new_dag = new ledger::DAG{db_name, load_on_start, certificate};
+    ret.reset(new_dag);
+  }
+
+  return ret;
```

Comment:
You are correct, however I did this because it is a shared pointer to a DAGInterface, the parent to DAG, so this reset casts the pointer upward implicitly

Reply:
That's fine, the cast will still take place on return. If you make it a std::make_shared<ledger::DAG>(...), it Should Just Work.